frontend/packages/operator-lifecycle-manager/src/components/operator-hub/operator-hub-page.tsx

```diff
@@ -37,6 +49,107 @@ const ANNOTATIONS_WITH_JSON = [
   OperatorHubCSVAnnotationKey.validSubscription,
 ];
 
+const clusterServiceVersionFor = (obj, clusterServiceVersions, version) => {
+  return clusterServiceVersions?.data?.items?.find(
+    (csv) => csv.metadata.name === `${obj.metadata.name}.${version}`,
+  );
+};
+
+const installPlanFor = (subscription, installPlans) => {
+  if (subscription) {
+    const installPlanName = subscription?.status?.installPlanRef?.name || '';
+    const installPlan: InstallPlanKind = installPlans?.data?.find(
+      (ip) => ip.metadata.name === installPlanName,
+    );
+    return installPlan;
+  }
+  return undefined;
+};
+
+const isOperatorInstalled = (
+  subscription,
+  operatorGroup,
+  packageName,
+  namespace,
+  clusterServiceVersion,
+  installPlan,
+) => {
```

Comment:
Also, as per my main review comment, this helper could be updated to be a hook that returns an install state string rather than a boolean, and we can actually just remove the `installed` property of OperatorHubItem. Wherever we use the `installed` property, we can use `installState` string comparisons or switch statement. This creates a single source of truth for installation state on the operator hub page.